### PR TITLE
Change wire repr

### PR DESCRIPTION
**Pull request by austingmhuang**
**Context:**
To accommodate the implementation of `qml.registers`, the `_repr_` of Wires should be something more copy-paste friendly rather than what it is currently

**Description of the Change:**
The `_repr_` of Wires is now `Wires([0])` rather than `<Wires = [0]>`

**Benefits:**
Easier to copy and paste

**Possible Drawbacks:**
Might make docs confusing in other places. Not sure where else <Wires = ...> notation will be found in docs.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by austingmhuang** [comment]
[sc-67920]

**Comment by trbromley** [comment]
Tagging @isaacdevlugt for visibility - not sure if this PR is ready to take a look at yet.

**Comment by austingmhuang** [comment]
> Tagging @isaacdevlugt for visibility - not sure if this PR is ready to take a look at yet.

It's ready! It's a very small PR that just addresses the changes that come about with changing the __repr__ of Wires, notably in the docs and in other tests.

**Review by isaacdevlugt — APPROVED** [review]
Nice!

File: pennylane/wires.py
Comment (soranjh) [inline pennylane/wires.py:1]: Please update other doc examples. This is a list I found:

```
pennylane/_device.py:        OrderedDict( [(<Wires = ['a']>, <Wires = [0]>), (<Wires = ['b']>, <Wires = [1]>)])
pennylane/pytrees/pytrees.py:    PyTree(AdjointOperation, (), [PyTree(RX, (<Wires = [0]>, ()), [Leaf])])
pennylane/pytrees/pytrees.py:    <PyTree(AdjointOperation, (), (<PyTree(Rot, (<Wires = [0]>, ()), (Leaf, Leaf, Leaf))>,))>
pennylane/_qubit_device.py:        <Wires = [0]>
pennylane/tape/tape.py:    <Wires = [0, 'a']>
pennylane/ops/op_math/controlled.py:    <Wires = [0, 1]>
pennylane/ops/op_math/controlled.py:    <Wires = [0]>
pennylane/ops/op_math/controlled.py:    <Wires = [1]>
pennylane/ops/qubit/hamiltonian.py:        {(1, frozenset({('PauliX', <Wires = [1]>, ()), ('PauliX', <Wires = [0]>, ())})),
pennylane/ops/qubit/hamiltonian.py:         (1, frozenset({('PauliZ', <Wires = [0]>, ())}))}
pennylane/workflow/return_types_spec.rst:<Wires = []>
pennylane/operation.py:        ((1.2,), (<Wires = [0, 1]>, (('pauli_word', 'XY'),)))
pennylane/operation.py:        (<Wires = ['b', 'c']>, (True, True), <Wires = []>))
pennylane/operation.py:        ((1.2, 2.3, 3.4), (<Wires = [0]>, ()))
pennylane/operation.py:        ((1.2,), (<Wires = [0, 1]>, (('pauli_word', 'XY'),)))
pennylane/operation.py:        {("PauliZ", <Wires = [1]>, ()), ("PauliX", <Wires = [0]>, ())}
pennylane/qchem/convert.py:    <Wires = [0, 1, 2]>
pennylane/qchem/convert.py:    <Wires = ['w0', 'w1', 'w2']>
pennylane/qchem/convert.py:    <Wires = ['w0', 'w1', 'w2']>
pennylane/qchem/convert.py:    <Wires = ['w0', 1, 'w2']>
pennylane/qchem/convert.py:    <Wires = ['w0', 'w1', 'w2']>
pennylane/qchem/openfermion_pyscf.py:    [<Wires = [0, 1, 2, 3, 4, 5]>, <Wires = [0, 1, 2, 3, 4, 5]>, <Wires = [0]>]
pennylane/templates/subroutines/trotter.py:        (<Wires = ['b', 'c']>, (True, True), <Wires = []>))
pennylane/templates/subroutines/trotter.py:        ((1.2, 2.3, 3.4), (<Wires = [0]>, ()))
pennylane/templates/subroutines/trotter.py:        ((1.2,), (<Wires = [0, 1]>, (('pauli_word', 'XY'),)))
pennylane/templates/subroutines/qubitization.py:        [AmplitudeEmbedding(array([1., 0.]), wires=[1]), Select(ops=(Z(0),), control=<Wires = [1]>), Adjoint(AmplitudeEmbedding(array([1., 0.]), wires=[1])), Reflection(, wires=[0])]
doc/development/plugins.rst:    print(wires[0]) # <Wires = ['auxiliary']>
doc/development/plugins.rst:    print(dev.wires) #  <Wires = ['q11', 'q12', 'q21', 'q22']>
doc/development/plugins.rst:    print(op.wires) # <Wires = ['q21', 'q11']>
doc/development/plugins.rst:    print(device_wires) # <Wires = [2, 0]>
doc/development/guide/architecture.rst:<Wires = ['a']>
doc/development/adding_operators.rst:   <Wires = ['a']>
doc/development/adding_operators.rst:    from (0.1,) and (<Wires = ['q3', 'q1']>, (('do_flip', True),)). You may need to override
```

**Review by soranjh — APPROVED** [review]

**Pull request merged**